src/main.rs

```diff
@@ -141,12 +153,15 @@ impl Config {
             index: matches.value_of("git")
                 .unwrap_or("https://github.com/rust-lang/crates.io-index.git")
                 .into(),
-            port: u16::from_str(matches.value_of("port")
-                    .unwrap_or("8080"))
-                .unwrap_or(8080),
-            refresh_rate: u64::from_str(matches.value_of("refresh")
-                    .unwrap_or("600"))
-                .unwrap_or(600),
+            port: port,
+            extern_url: matches.value_of("extern-url")
+                .map(Into::into)
+                .unwrap_or(format!("http://localhost:{}", port)),
+            refresh_interval: matches.value_of("refresh")
```

Comment:
I'd rather have this parse handled before the struct definition because it does unwrap if the format of "-r" is invalid:

```
cargo run -- -r 3t
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `target/debug/cargo-cacher -r 3t`
thread 'main' panicked at 'parse refresh interval: UnknownUnit(1, 2)', src/libcore/result.rs:1084:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

If it's handled non-inline, the error can be handled and a nicer error logged before bailing

Reply:
Do you think this is nice enough?
```
     Running `target/debug/cargo-cacher -r 3t`
Error while parsing refresh interval: unknown unit at 1-2.
Try values like "600s" or "2 hours".
```
Just printing out the string positions like that isn't exactly beautiful, but if possible, I'd like to improve this in the humantime crate.

Reply:
Love it :)